bravado/swagger_model.py

```diff
@@ -24,22 +27,25 @@ class FileEventual(object):
     class FileResponse(object):
 
         def __init__(self, data):
-            self.data = data
+            self.text = data
+            self.headers = {}
 
         def json(self):
-            return self.data
+            return self.text
 
     def __init__(self, path):
         self.path = path
+        self.is_yaml = is_yaml(path)
 
     def get_path(self):
-        if not self.path.endswith('.json'):
+        if not self.path.endswith('.json') and not self.is_yaml:
             return self.path + '.json'
         return self.path
 
     def wait(self, timeout=None):
         with contextlib.closing(urllib.request.urlopen(self.get_path())) as fp:
-            return self.FileResponse(json.load(fp))
+            content = fp.read() if self.is_yaml else json.load(fp)
```

Comment:
It's because it's mimicking a response object, which has a json() method to return a Python representation of JSON content. So in Loader.load_spec we distinguish between the two cases and either load YAML ourselves or just call response.json().

The FileResponse object was originally built for the JSON case, so it simply parsed JSON immediately and then returned the parsed data when json() is called. I'm going to get rid of this and separate out the two steps of reading JSON content from disk and then parsing it.